File: arm-fragment/core/src/main/java/com/iur/arm/fragment/core/exception/AfterSaveStateTransactionWarning.java
package com.iur.arm.fragment.core.exception;

public class AfterSaveStateTransactionWarning extends RuntimeException {

    public AfterSaveStateTransactionWarning(String action) {
        super("Warning: Perform this " + action + " action after onSaveInstanceState!");
        Timber.d(getMessage());
    }

}

File: arm-fragment/core/src/main/java/io/github/iur/arm/fragment/core/exception/AfterSaveStateTransactionWarning.kt
package io.github.iur.arm.fragment.core.exception

import timber.log.Timber

class AfterSaveStateTransactionWarning : RuntimeException {
    constructor(action: String) : super("Warning: Perform this $action action after onSaveInstanceState!")

    init {
        Timber.d(message)
    }
}
